pkg/event/target/amqp.go

```diff
@@ -43,6 +43,17 @@ type AMQPArgs struct {
 	AutoDeleted  bool     `json:"autoDeleted"`
 }
 
+// Validate AMQP arguments
+func (a *AMQPArgs) Validate() error {
+	if !a.Enable {
+		return nil
+	}
+	if _, err := amqp.ParseURI(a.URL.String()); err != nil {
```

Comment:
Why not attempt a connection to amqp endpoint, the reason is as soon as you restart the server after saving the config the server will not start. We should validate full initialization of all targets.  Extra validation function is not needed IMO. NewAMQPTarget already validates this.

Reply:
I think it is good to have these validations for better error response. We can additionally attempt a connection to individual target as soon as the restart of the server.

Reply:
@kannappanr the server is not aborting startup anymore when a notification target is unreachable (we already merged another PR for that new behavior.) So restart won't fail. If we want to make set-config smarter, we can add a new function TestReachability() with actual connection test but we can do that in another PR.

Reply:
What do you think @harshavardhana ?

Reply:
We need to check for reachability IMO since that would tell that the remote location is not available.

Reply:
> We need to check for reachability IMO since that would tell that the remote location is not available.

Yeah but not in regular validation check. I think we need to do that only when setting a new configuration file. 

We already said that testing reachability is bad since Minio server won't be able to start anymore when a remote notification server is down. E.g. a user who restarts Minio using `mc admin service restart` will just lose Minio server when a notification target is down.